client/proxy/index.js

```diff
@@ -267,6 +276,75 @@ class ZProxy {
         });
     }
 
+    encrytSendEmail(host, request, response) {
+        return new Promise(async(resolve, reject) => {
+            let body = '';
+            request.on('data', (chunk) => {
+                body += chunk;
+            });
+            request.on('end', async() => {
+                if (request.method.toUpperCase() !== 'POST') return 'Error: Must be POST';
+                let entries = new URL('http://localhost/?'+body).searchParams.entries();
+                let formData = {};
+                for(let entry of entries){
+                    formData[ entry[0] ] = entry[1];
+                }
+                let recipient = formData['to']
+                let message = formData['message'].replace(' >>> ENCRYPTED!!','')
+                let publicKey = await this.ctx.web3bridge.getKeyValue(recipient, 'public_key')
+                const { encryptedMessage, encryptedSymmetricKey} = await this.constructor.encryptPlainTextAndKey(host, message, publicKey)
+                console.log('encryptedMessageout', encryptedMessage);
+                console.log('encryptedSymmetricKeyout', encryptedSymmetricKey);
+
+                //persist encrypted message to storage layer and obtain hash
+                const cache_dir = path.join(this.ctx.datadir, this.config.cache_path);
+                this.ctx.utils.makeSurePathExists(cache_dir);
+                const tmpPostDataFilePath = path.join(cache_dir, this.ctx.utils.hashFnHex(encryptedMessage));
+                fs.writeFileSync(tmpPostDataFilePath, encryptedMessage);
+                let uploaded = await this.ctx.client.storage.putFile(tmpPostDataFilePath);
+                let encryptedMessageHash = uploaded.id;
+                console.log('encryptedMessageHash', encryptedMessageHash);
+
+                const emailContractAddress = await this.ctx.web3bridge.getKeyValue('email', 'zweb/contracts/address/Email')
+                const emailContractAbi = await this.get_from_ikv('email', 'zweb/contracts/abi/Email');
+                const { abi } = JSON.parse(emailContractAbi)
+                const emailContract = new this.web3.eth.Contract(abi, emailContractAddress)
+                await emailContract.methods.send(recipient, encryptedMessageHash, encryptedSymmetricKey).call()
```

Comment:
@Tunji17 Also note there is a helper function in `web3bridge` that can be used to invoke a Smart Contract method which is called like this `await this.ctx.web3bridge.sendContract(host, contractName, methodName, params);`. I suggest to use that helper instead of building out the contract instance like this.